specs/sb/api.md

```diff
@@ -342,6 +400,45 @@ A web-friendly display name is camel-cased with spaces and punctuation supported
         }, {
           "content": "40 concurrent connections"
         }]
+      },
+      "schemas": {
+        "service_instance": {
+          "parameters": {
+            "required": [
+              "fake_required_property"
+            ],
+            "properties": {
+              "fake_required_property": {
+                "type": "string"
+              }
+            }
+          },
+          "outputs": {
+            "properties": {
```

Comment:
I would suggest PRs representing proposed changes to the broker API spec address individual use cases so they can be discussed discreetly. This PR seems to conflate several use cases. Best I can tell there are several things going on in this PR:
- UI clients should be able to dynamically display UI elements for capturing inputs to construct the `parameters` field which can be optionally sent with the `PUT /v2/service_instances/:guid`, `PATCH /v2/service_instances/:guid`, and `PUT /v2/service_instances/:guid/service_instances/:guid` endpoints.
- Some use case related to outputs for service bindings. I don't understand what the use case is, but it is IMO independent from the first one.

> However it is within scope to schematize outputs for service bindings, 
Schematizing outputs for service bindings is a separate use case from schematizing inputs for the parameters field. The former is not well defined and IMO needs further discussion. The latter is more well defined and actionable. Conflating the two is confusing. 

> and I didn't want to define separate objects for each. I can either separate the objects into nearly identical objects, or specify in the docs that outputs schema is only valid for bindings. How do these approaches seem to you?
I don't understand what you're referring to.